chain-core/src/init/coin.rs

```diff
@@ -241,6 +241,39 @@ mod test {
     use quickcheck::quickcheck;
 
     quickcheck! {
+        // test whether oveflow error occur
+        fn coin_add() -> bool {
+            let a = Coin::max();
+            let b = Coin::max();
+            let sum = a+ b;
+            sum.is_err()
+        }
+
+        // test whether overflow error not occur
+          fn coin_add2() -> bool {
+            let a = Coin::max();
+            let b = Coin::new(0).unwrap();
```

Comment:
could this actually cause overflows?
The "lowest" overflowing addition is: Coin::max() + Coin::new(84_467_440_737__0955_1616), I think?

Reply:
this is not overflow.
this is assert for "max + 0" works fine

Reply:
ok, you may want to add a test case with `Coin::new(84_467_440_737__0955_1616)`